frontend/source/components/root/root.vue

```diff
@@ -71,20 +72,26 @@ function sendDomEvent(eventName, detail = {}) {
 	}));
 }
 
-const onExitTutorial = () => {
-	isTutorialClosed.value = true;
+const updateTutorialState = (actionType) => {
+	const currentPathname = window.location.pathname;
 	const state = loadStateFromLocalStorage();
-	state.isClosed = true;
+	const tutorialState = state[currentPathname] || {steps: []};
+	tutorialState[actionType] = true;
+	state[currentPathname] = tutorialState;
 	saveStateToLocalStorage(state);
-	sendDomEvent('total-tutorial-exit', {step: config.currentStep});
+	if (actionType === 'closed' || actionType === 'finished') {
```

Comment:
you call it "isFinished" but here you're looking at something else ("finished"). are you sure that's right?